### ISSUE TITLE
dry run fails on re.compile when software name contains metacharacters like '++'

### ISSUE THREAD

**Issue by migueldiascosta** [hpcugent/easybuild-framework#2097]
e.g. 

```
# eb Blitz++-0.10-foss-2016a.eb -x
(...)
ERROR: Traceback (most recent call last):
  File "easybuild-framework/easybuild/main.py", line 116, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "easybuild-framework/easybuild/framework/easyblock.py", line 2487, in build_and_install_one
    app = app_class(ecdict['ec'])
  File "easybuild-framework/easybuild/framework/easyblock.py", line 244, in __init__
    self.init_dry_run()
  File "easybuild-framework/easybuild/framework/easyblock.py", line 291, in init_dry_run
    dry_run_set_dirs(tmp_root_dir, self.builddir, self.installdir, self.installdir_mod)
  File "easybuild-framework/easybuild/tools/build_log.py", line 245, in dry_run_set_dirs
    DRY_RUN_SOFTWARE_INSTALL_DIR = (re.compile(software_installdir), software_installdir[len(prefix):])
  File "/usr/lib64/python2.6/re.py", line 190, in compile
    return _compile(pattern, flags)
  File "/usr/lib64/python2.6/re.py", line 245, in _compile
    raise error, v # invalid expression
error: multiple repeat
```

needs `re.escape`?

**Comment by boegel** [comment]
@migueldiascosta Nice catch...

`re.escape` seems like the easiest/best way to fix this problem, yes

Are you up for issuing a PR for that?

**Comment by migueldiascosta** [comment]
@boegel ok, I'll replace `re.compile(software_installdir)` with `re.compile(re.escape(software_installdir))` 

`re.compile(builddir)` seems to be safe by construction (`remove_unwanted_chars` in `gen_builddir`)

regarding `re.compile(module_installdir)`, I suppose one wouldn't expect those characters in `module_installdir`, but shall I escape it anyway to be safe?

**Comment by boegel** [comment]
@migueldiascosta I'd keep things consistent and use `re.compile(re.escape(...))` in all three cases, it doesn't hurt (just keep in mind to stick to lines of <= 120 chars)

**Comment by boegel** [comment]
fixed in #2098
